tutorials/22_PromptNode_with_Shaper.ipynb

```diff
@@ -0,0 +1,337 @@
+   "metadata": {},
+   "source": [
+    "> `InMemoryDocumentStore` is the simplest DocumentStore to get started with. It requires no external dependencies and it's a good option for smaller projects and debugging. But it doesn't scale up so well to larger Document collections, so it's not a good choice for production systems. To learn more about the DocumentStore and the different types of external databases that we support, see [DocumentStore](https://docs.haystack.deepset.ai/docs/document_store)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The DocumentStore is now ready. Now it's time to fill it with some Documents."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Fetching and Writing Documents\n",
+    "\n",
+    "We'll use the Wikipedia pages of [Seven Wonders of the Ancient World](https://en.wikipedia.org/wiki/Wonders_of_the_World) as Documents. We preprocessed the data and uploaded to a Hugging Face Space: [Seven Wonders](https://huggingface.co/datasets/bilgeyucel/seven-wonders). Thus, we don't need perform any additional cleaning or splitting. \n",
+    "\n",
+    "Let's fetch the data and write it to the DocumentStore: "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from datasets import load_dataset\n",
+    "\n",
+    "dataset = load_dataset(\"bilgeyucel/seven-wonders\", split=\"train\")\n",
+    "\n",
+    "document_store.write_documents(dataset)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Initializing the Retriever\n",
+    "\n",
+    "Let's initialize a BM25Retriever and make it use the InMemoryDocumentStore we initialized earlier in this tutorial:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from haystack.nodes import BM25Retriever\n",
+    "\n",
+    "retriever = BM25Retriever(document_store=document_store, top_k=2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Initializing the Shaper\n",
+    "\n",
+    "[Shaper](https://docs.haystack.deepset.ai/docs/shaper) is necessary when the output of one node differs from the format of the next node. In our pipeline, we need to join retrieved Documents in a way that we can inject these Documents into the prompt. Shaper can solve this problem with [`join_documents`](https://docs.haystack.deepset.ai/reference/shaper-api#join_documents) function. For Shaper, the keys of `inputs` will refer to parameters for its function and the values of `inputs` will be the arguments of the function. Retriever refers Documents as `documents` and `join_documents` expects `documents` parameter, so, we can pass `{\"documents\": \"documents\"}` to the Shaper as `inputs`. To output joined Documents as `texts`, we need to define  `outputs=[\"texts\"]`.    \n",
```

Comment:
```suggestion
    "[Shaper](https://docs.haystack.deepset.ai/docs/shaper) is necessary when the output of one node differs from the format of the next node. In our pipeline, we need to join retrieved Documents so that we can inject these Documents into the prompt. Shaper can solve this problem with the [`join_documents`](https://docs.haystack.deepset.ai/reference/shaper-api#join_documents) function. For Shaper, the keys of `inputs` will refer to parameters for its function and the values of `inputs` will be the arguments of the function. Retriever refers Documents as `documents` and `join_documents` expects `documents` parameter, so, we can pass `{\"documents\": \"documents\"}` to the Shaper as `inputs`. To output joined Documents as `texts`, we need to define  `outputs=[\"texts\"]`.    \n",
```